test/metabase/query_processor_test/filter_test.clj

```diff
@@ -109,6 +110,46 @@
                    {:aggregation [[:count]]
                     :filter      [:between [:datetime-field $date :day] "2015-04-01" "2015-05-01"]}))))))))
 
+(defn- mongo-major-version [db]
+  (when (= driver/*driver* :mongo)
+    (-> (driver/describe-database :mongo db)
+        :version (str/split #"\.") first parse-long)))
+
+(deftest temporal-arithmetic-test
+  (testing "Should be able to use temporal arithmetic expressions in a `:between` filter (#22531)"
+    ;; we also want to test this against MongoDB but [[mt/normal-drivers-with-feature]] would normally not include that
+    ;; since MongoDB only supports expressions if version is 4.0 or above and [[mt/normal-drivers-with-feature]]
+    ;; currently uses [[driver/supports?]] rather than [[driver/database-supports?]] (TODO FIXME)
+    (mt/test-drivers (conj (mt/normal-drivers-with-feature :expressions) :mongo)
+      (mt/dataset attempted-murders
```

Comment:
Actually MongoDB 3.x was EOL'ed over a year ago so at this point it's probably safe to say we don't support MongoDB 3.x anymore. We don't test against it anyway -- we test against 4.0 and 5.0 IIRC. You could actually just replace that `database-supports?` method for `[:mongo :expressions]` with a `supports?` method and then we won't need this special case code here if you want. That would give us the added benefit of having other `:expressions` code run against Mongo without us needing special case code here. Might catch some more bugs

Reply:
That seems to provoke quite some test failures so I prefer to do it separately, not in this PR.